Add a new color customization How-to page to PWA docs (#24925)
mdn/content: files/en-us/web/progressive_web_apps/how_to/customize_your_app_colors/index.md

---
title: Customize your app's theme and background colors
slug: Web/Progressive_web_apps/How_to/Customize_your_app_colors
---

> When building Progressive Web Apps (PWAs), it's important to not only consider the appearance of the content your app displays but also the way the app appears on the user's device once the app is installed. One way to customize the window in which your app appears is by using the [`theme_color`](/en-US/docs/Web/Manifest/theme_color) and [`background_color`](/en-US/docs/Web/Manifest/background_color) [web app manifest](/en-US/docs/Web/Manifest) members.

Reviewer: ```suggestion
When building Progressive Web Apps (PWAs), it's important to not only consider the appearance of your app's content, but also the way the app appears on the user's device once the app is installed. One way to customize the window in which your app appears is by using the [`theme_color`](/en-US/docs/Web/Manifest/theme_color) and [`background_color`](/en-US/docs/Web/Manifest/background_color) [web app manifest](/en-US/docs/Web/Manifest) members.
```

> By using the `theme_color` and `background_color` members in your PWA manifest, you can create a more polished experience for your users. These small details can help make your PWA feel more like an OS-native app, and more familiar to your users.

Reviewer: ```suggestion
By defining the `theme_color` and `background_color` members in your PWA manifest, you can create a more polished experience for your users. These small details can help make your PWA feel more like an OS-native app, and more familiar to your users.
```

## Customize the app window background color

Use the [`background_color`](/en-US/docs/Web/Manifest/background_color) manifest member in your PWA to define the color that appears in the application window before the app's stylesheets have loaded.

The color you choose should match the `background-color` CSS property in your app's stylesheet for a smooth transition between launching the web application and loading its content.

Here is a web app manifest file example, showing how to define a background color:

```json
{
  "name": "My app",
  "start_url": "/",
  "icons": [
    {
      "src": "icon.webp",
      "sizes": "48x48",
      "type": "image/webp"
    }
  ],
  "display": "standalone",

>   "background_color: "peachpuff"

Reviewer 2: ```suggestion
  "background_color": "peachpuff"
```

}
```

## Define a theme color

In addition to customizing the background color of your app window, you can use the [`theme_color`](/en-US/docs/Web/Manifest/theme_color) member in your PWA manifest to define the color of certain operating system and browser UI elements that your PWA may use.

Different devices, operating systems, and browsers apply the `theme_color` member differently. For example:

- On Android devices, the theme color is applied to the browser UI and task switcher background.
- On Windows, the theme color is used to customize the title bar of your standalone app window.

Choose a `theme_color` that complements the overall design and branding of your PWA and works well across a variety of devices and platforms to ensure a consistent user experience.

Here is a web app manifest file example, showing how to define a theme color:

```json
{
  "name": "My app",
  "start_url": "/",
  "icons": [
    {
      "src": "icon.webp",
      "sizes": "48x48",
      "type": "image/webp"
    }
  ],
  "display": "standalone",

>   "theme_color: "peachpuff"

Reviewer 2: ```suggestion
  "theme_color": "peachpuff"
```

}
```

### Relationship with the `theme-color` meta element value

> The [`theme-color`](/en-US/docs/Web/HTML/Element/meta/name/theme-color) value for the {{htmlattrxref("name", "meta")}} attribute of the {{htmlelement("meta")}} HTML element can be used to define a theme color per webpage. This is different from the `theme_color` manifest member which you define only once, globally for your app.
> 
> If both are set, the `theme-color` meta element value overrides the `theme_color` manifest member. This lets you define a global color for your app, and override it on specific pages.
> 
> Note that on certain browsers like Safari on macOS, or Chrome on Android, the `theme-color` meta element value is used to style the browser UI too.

Reviewer: ```suggestion
The [<meta>](/en-US/docs/Web/HTML/Element/meta) element, used to provide document metadata in terms of [name](/en-US/docs/Web/HTML/Element/meta#name)-value pairs, can also be used to set the theme color on a per-page basis. 

The [`theme-color`](/en-US/docs/Web/HTML/Element/meta/name/theme-color) value for the `name` attribute defines the theme color for the page on which it is set. This is different from the `theme_color` manifest member which you define only once, globally for your app.

with the name attribute giving the metadata name, and the [content](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta#content) attribute giving the value.

If both are set, the `theme-color` meta value overrides the `theme_color` manifest member. This lets you define a global color for your app, and override it on individual pages.

Note that for certain browsers, like Safari on macOS and Chrome on Android, the `theme-color` meta element value sets the theme color for the browser UI as well.
```

Reviewer: htmlattrxref is being deprecated i think

Choose a `theme_color` that complements the overall design and branding of your PWA and works well across a variety of devices and platforms to ensure a consistent user experience.

As with `background_color`, any [`<color>`](/en-US/docs/Web/CSS/color_value) value is valid. In this web app manifest file example, the `theme_color` is set to the `rgb(255 218 185)`, the [RGB](/en-US/docs/Web/CSS/color_value/rgb) equivalent of `peachpuff`:

```json
{
  "name": "My app",
  "start_url": "/",
  "icons": [
    {
      "src": "icon.webp",
      "sizes": "48x48",
      "type": "image/webp"
    }
  ],
  "display": "standalone",
  "background_color": "peachpuff",
  "theme_color": "#bd5a1e"
}
```

The following screenshot shows the above code in action when the app is installed on Windows, where the `theme_color` manifest member is used as the color of the title bar:

![The app window, on Windows, showing the theme color in the title bar](./background-theme-colors-windows.png)

### Relationship with the `theme-color` meta element value

> The [`theme-color`](/en-US/docs/Web/HTML/Element/meta/name/theme-color) value for the [`name`](/en-US/docs/Web/HTML/Element/meta/name) attribute of the {{htmlelement("meta")}} HTML element can be used to define a theme color per webpage. This is different from the `theme_color` manifest member which you define only once, globally for your app.

Reviewer: ```suggestion
The [`theme-color`](/en-US/docs/Web/HTML/Element/meta/name/theme-color) value for the [`name`](/en-US/docs/Web/HTML/Element/meta/name) attribute of the {{htmlelement("meta")}} HTML element can be used to define a theme color per webpage. This is different from the `theme_color` manifest member which you define only once, globally, for your app.
```
